nuxeo-management-rest-api/src/main/java/org/nuxeo/rest/management/elasticsearch/ElasticsearchObject.java

```diff
@@ -0,0 +1,170 @@
+
+    /**
+     * Executes an ES flush on document index of a given repository.
+     */
+    @POST
+    @Path("{repositoryName}/flush")
+    public void doFlush(@PathParam("repositoryName") String repositoryName) {
+        Framework.doPrivileged(
+                () -> Framework.getService(ElasticSearchAdmin.class).flushRepositoryIndex(repositoryName));
+    }
+
+    /**
+     * Executes an ES optimize on document index of a given repository.
+     */
+    @POST
+    @Path("{repositoryName}/optimize")
+    public void doOptimize(@PathParam("repositoryName") String repositoryName) {
+        Framework.doPrivileged(
+                () -> Framework.getService(ElasticSearchAdmin.class).optimizeRepositoryIndex(repositoryName));
+    }
+
+    /**
+     * Executes an ES flush on all documents indexes.
+     */
+    @POST
+    @Path("flush")
+    public void doFlush() {
+        Framework.doPrivileged(() -> Framework.getService(ElasticSearchAdmin.class).flush());
+    }
+
+    /**
+     * Executes an ES optimize on all documents indexes.
+     */
+    @POST
+    @Path("optimize")
+    public void doOptimize() {
+        Framework.doPrivileged(() -> Framework.getService(ElasticSearchAdmin.class).optimize());
+    }
+
+    /**
+     * Performs an ES indexing on documents of the given repository and that match the NXQL query.
+     *
+     * @param repositoryName the repository name on which we launch the indexing, cannot be {@code null}
+     * @param query the NXQL that documents must match to be indexed, can be {@code null} and in this case all documents
+     *            of the given repository will be indexed
+     * @return the http response containing the location of the launched indexing
+     */
+
+    protected Response performIndexing(String repositoryName, String query) {
+        RepositoryService repositoryService = Framework.getService(RepositoryService.class);
+        if (repositoryService.getRepository(repositoryName) == null) {
+            throw new NuxeoException(String.format("No such repository with name: %s", repositoryName), SC_NOT_FOUND);
+        }
+
+        String nxql = StringUtils.defaultIfBlank(query, "Select * From Document");
+        BulkService bulkService = Framework.getService(BulkService.class);
+        String indexingId = bulkService.submit(
+                new BulkCommand.Builder(ACTION_NAME, nxql).repository(repositoryName)
+                                                          .param(INDEX_UPDATE_ALIAS_PARAM, true)
+                                                          .user(SYSTEM_USERNAME)
+                                                          .build());
+
+        URI statusLocation = UriBuilder.fromPath(ctx.getBaseURL())
+                                       .path(ctx.getBasePath())
+                                       .path("/elasticsearch/indexing/{indexingId}/status")
+                                       .build(indexingId);
+        return Response.created(statusLocation).build();
```

Comment:
Here it would be interesting to follow what was introduced in `RenditionsObject`. Ie: return the bulk status with an OK http status. Here the created location stuff can be error prone, `ctx.getBaseURL()` can return something which is not the public URI (for instance when nuxeo is behind a proxy) and this forces us to keep synchronize the path.